File: check_prometheus.js
const http = require('http');

const options = {
  hostname: 'udb-prometheus',
  port: 9090,
  path: '/api/v1/targets',
  method: 'GET'
};

const req = http.request(options, (res) => {
  let data = '';
  res.on('data', chunk => data += chunk);
  res.on('end', () => {
    try {
      const parsed = JSON.parse(data);
      const targets = parsed.data.activeTargets;
      const up = targets.filter(t => t.health === 'up').length;
      console.log('UP: ' + up + '/' + targets.length);
      targets.forEach(t => {
        console.log('  ' + (t.labels.job || '?') + ': ' + t.health);
      });
    } catch (e) {
      console.error('Parse error:', e.message);
    }
  });
});

req.on('error', (e) => console.error('Error:', e.message));
req.end();

File: check_prometheus.py
import sys, json, urllib.request
url = "http://udb-prometheus:9090/api/v1/targets"
with urllib.request.urlopen(url) as response:
    data = json.load(response)
    up = sum(1 for t in data['data']['activeTargets'] if t['health'] == 'up')
    total = len(data['data']['activeTargets'])
    print(f"UP: {up}/{total}")
    for t in data['data']['activeTargets']:
        print(f"  {t['labels'].get('job','?')}: {t['health']}")
